holidays/utils.py

```diff
@@ -281,13 +281,13 @@ def list_supported_countries() -> Dict[str, List[str]]:
         the value is a list of supported subdivision codes.
     """
     return {
-        cls.country: cls.subdivisions
+        cls.country if unique else name: cls.subdivisions
```

Comment:
@arkid15r this code generates a Set when unique is set, which does not conform to the declared return type.

Reply:
Couldn't reproduce that:
```python
>>> import holidays
>>> holidays.__version__
'0.22'
>>> type(holidays.list_supported_countries(unique=True))
<class 'dict'>
```